### DISCUSSION TITLE
The new form helper

### DISCUSSION THREAD

**Discussion by captenmasin** [inertiajs/inertia#429] [Help] [answered]
Does the new form helper have resetOnSuccess and preserveScroll?

I know it was kind of based on the Jetstream form helper which does have those options, but seems the official helper doesn't have these?

**Comment by claudiodekker** [comment, marked as answer]
Hi,

That's correct. Inertia's first-party form does not have `resetOnSuccess`, as it caused a lot of issues amongst some our users, as well as that we considered it to be a somewhat counter-intuitive feature. Here's why: When an Inertia request finishes, Inertia's state is already updated automatically. Because of this, 'resetting' it on success wouldn't do anything, because the values you've entered are already in the 'target' state!

Of course, there are exclusions to this, such as a 'change password' form: When the request succeeds, you want to clear any password the user might have entered. For this reason, we have `this.form.reset()` / `this.form.reset(field, field2, ...)`, which you can simply chain onto an `onSuccess` event callback:
```js
const form = this.$inertia.form({ .. });

form.post('endpoint', {
   preserveScroll: true,
   onSuccess: () => form.reset('password', 'password_confirmation')
});
```

As you might already be able to tell from the example above, `preserveScroll` is totally available (and in fact, most if not all other Inertia request options are!), and applying them is as simple as just passing them through the request options.

Hope this helps!

**Reply by captenmasin** [reply]
Ooooh that makes so much sense!

Thank you for this
